src/solidity/decodeInfo.js

```diff
@@ -0,0 +1,353 @@
+    storageBytes: 1,
+    typeName: extracted[0] + ' ' + extracted[1],
+    enum: getEnum(type, stateDefinitions),
+    name: 'enum'
+  }
+}
+
+/**
+  * Struct decode the given @arg type
+  *
+  * @param {String} type - type given by the AST
+  * @return {Object} returns decoded info about the current type: { needsFreeStorageSlot, storageBytes, typeName, decoder, members}
+  */
+function Struct (type, stateDefinitions) {
+  var extracted = type.split(' ')
+  type = extracted[0] + ' ' + extracted[1]
+  var membersDetails = getStructMembers(type, stateDefinitions)
+  return {
+    needsFreeStorageSlot: true,
+    storageBytes: membersDetails.storageBytes,
+    typeName: type,
+    members: membersDetails.members,
+    name: 'struct'
+  }
+}
+
+/**
+  * retrieve enum declaration of the given @arg type
+  *
+  * @param {String} type - type given by the AST
+  * @param {Object} stateDefinitions  - all state declarations given by the AST (including struct and enum type declaration)
+  * @return {Array} - containing all value declaration of the current enum type
+  */
+function getEnum (type, stateDefinitions) {
+  for (var k in stateDefinitions) {
+    var dec = stateDefinitions[k]
+    if (dec.name === 'EnumDefinition' && type.indexOf('enum ' + dec.attributes.name) === 0) {
+      return dec.children
+    }
+  }
+  return null
+}
+
+/**
+  * retrieve memebers declared in the given @arg tye
+  *
+  * @param {String} typeName - name of the struct type (e.g struct <name>)
+  * @param {Object} stateDefinitions  - all state definition given by the AST (including struct and enum type declaration)
+  * @return {Array} containing all members of the current struct type
+  */
+function getStructMembers (typeName, stateDefinitions) {
+  var members = []
+  var storageBytes = 0
+  for (var k in stateDefinitions) {
+    var dec = stateDefinitions[k]
+    if (dec.name === 'StructDefinition' && typeName.indexOf('struct ' + dec.attributes.name) === 0) {
+      for (var i in dec.children) {
+        var member = dec.children[i]
+        var decoded = decode(member.attributes.type, stateDefinitions)
+        members.push(decoded)
+        if (decoded.needsFreeStorageSlot) {
+          storageBytes = Math.ceil(storageBytes / 32) * 32
+        }
+        storageBytes += decoded.storageBytes
+      }
+      break
+    }
+  }
+  return {
+    members: members,
+    storageBytes: storageBytes
+  }
+}
+
+/**
+  * return the size of the current array
+  *
+  * @param {String} typeName - short type ( e.g uint[][4] )
+  * @return {String|Int} return 'dynamic' if dynamic array | return size of the array
+  */
+function extractArraySize (typeName) {
+  if (typeName.indexOf('[') !== -1) {
+    var squareBracket = /\[([0-9]+|\s*)\]/g
+    var dim = typeName.match(squareBracket)
+    var size = dim[dim.length - 1]
+    if (size === '[]') {
+      return 'dynamic'
+    } else {
+      return parseInt(dim[dim.length - 1].replace('[', '').replace(']'))
```

Comment:
You can use the regex match expression here to get the part inside the square brackets.